ModularTegustation/ego_weapons/melee/DeadRabbitsBats.dm

```diff
@@ -0,0 +1,103 @@
+/obj/item/ego_weapon/city/DRabbitbat//it's just a HE W.corp baton that deals red
+	name = "Dead Rabbits Bat"
+	desc = "A nasty-looking bat covered with nails."
+	special = "This weapon has a combo system with a stun finisher."
+	icon = 'ModularTegustation/Teguicons/DeadRabbitBatIcons.dmi'
+	icon_state = "RabbitBat"
+	lefthand_file = 'ModularTegustation/Teguicons/DeadRabbitBatsL.dmi'
+	righthand_file = 'ModularTegustation/Teguicons/DeadRabbitBatsR.dmi'
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	force = 25
+	damtype = RED_DAMAGE
+	attack_verb_continuous = list("bashes", "crushes")
+	attack_verb_simple = list("bash", "crush")
+	attribute_requirements = list(
+							FORTITUDE_ATTRIBUTE = 80
+							)
+	var/combo = 0
+	/// Maximum world.time after which combo is reset
+	var/combo_time
+	/// Wait time between attacks for combo to reset
+	var/combo_wait = 20
+	var/waltz_partner

```

Comment:
this is a hard-delete waiting to happen, you should clear the reference on `Destroy()` and maybe attach a signal onto whoever you mark so if they start deleting you clear the reference to them